controller/src/main/java/io/pravega/controller/server/rpc/auth/GrpcAuthHelper.java

```diff
@@ -75,7 +75,13 @@ public String checkAuthorization(String resource, AuthHandler.Permissions expect
 
     public String checkAuthorizationAndCreateToken(String resource, AuthHandler.Permissions expectedLevel) {
         if (isAuthEnabled) {
-            checkAuthorization(resource, expectedLevel);
+            try {
+                checkAuthorization(resource, expectedLevel);
+            } catch (RuntimeException e) {
+                // An auth handler plugin loaded in the system may throw this exception.
+                log.warn("Authorization failed", e);
```

Comment:
Is this actionable? The main difference between warn and error is that warn is something that the user should know, but it is not necessarily actionable, while an error typically indicates an unrecoverable situation that requires action from the user.

I see that you used warn for all new messages. Are they really warnings according to the definition above?

Reply:
It is not actionable at this point. A custom auth handler may throw an exception and we want to have a log of it. This may be a temporary issue if the custom Auth Handler talks to an Identity and Access Management system. Accordingly, I think `WARN` level is appropriate in this case.

The same is the case for other log statements I've added. I have done checks for whether auth is enabled, and if it is enabled, a null or empty delegation token hints to a potential problem.